via/services/secure_link.py

```diff
@@ -0,0 +1,39 @@
+from h_vialib.exceptions import TokenException
+from h_vialib.secure import ViaSecureURL
+from pyramid.httpexceptions import HTTPUnauthorized
+
+
+def has_secure_url_token(view):
+    """Require the request to have a valid signature."""
+
+    def view_wrapper(context, request):
+        if not request.find_service(SecureLinkService).request_is_valid(request):
+            return HTTPUnauthorized()
+
+        return view(context, request)
+
+    return view_wrapper
```

Comment:
:+1:

As a view decorator the previous `views/decorators.py` location seems like the "right" place for this: `services/` is supposed to be for "business logic"-type concerns and decoupled from Pyramid (as much as possible) whereas `views/` is supposed to be for views-like concerns and coupled to Pyramid, and a view decorator is very much a Pyramid/views concern: it exists only to serve Pyramid, to be used by Pyramid `@view_config`'s, the entire definition of what a view decorator _is_, how it gets called, with what arguments, what it returns, is determined by Pyramid.

That said, I don't think the "violation" here (of the definition of the services directory) is a big deal and it keeps `SecureLinkService`-related code together.

Reply:
The code in this decorator seems to be simpler than [how it used to be](https://github.com/hypothesis/via/blob/b4839541d21ba63a05fccf8f6334ec7dd58da5ce/via/views/decorators.py#L7-L23). It's no longer checking the `signed_urls_required` setting and I can see that got moved into `SecureLinkService.request_is_valid()` below. It's also no longer pulling the signature out from `request.params.get(signature_param)` but I don't see where that went?

Reply:
The param is extracted and checked inside the `ViaSecureURL`.

There is a change of behavior here. Previously this would allow you skip signature checking if there was no signature, _and_ no signature was required. Now we just skip checking if no signature is required. This is why it's gotten simpler and now clearer that us getting the param out wasn't really for much. Having a variable for the param name was misleading too, as it implied you could change the value, but the code didn't pass this through to the `ViaSecureURL` as we should have and so you can't really.

I think this previous behavior was to allow us to deploy it and test it without it being enabled. So we _needed_ it to fail for testing purposes, not really security purposes.